### PR TITLE
U532/t551 replace event dispatcher

### PR DESCRIPTION
**Pull request by Smaugmuel**
Sooooo many files changed.
Replaced old event dispatcher in Application with a new one.
Fixed issue where water shot after the last player died was not removed correctly.

### CODE REVIEW

File: SPLASH/src/game/events/NetworkWelcomeEvent.h [hunk 1]
```diff
@@ -1,27 +1,13 @@
...
-	~NetworkWelcomeEvent() {}
-
-	inline std::list<Player> getListOfPlayers() { return playerList; }
-
-	static Type GetStaticType() {
-		return Event::NETWORK_WELCOME;
-	}
-
-private:
-	std::list<Player> playerList;
```
Comment (OliverGlandberger) [inline SPLASH/src/game/events/NetworkWelcomeEvent.h]: m_playerList? No big deal, but just sayin'. ^^

File: Sail/src/API/DX12/dxr/DXRBase.cpp [hunk 2]
```diff
@@ -394,20 +403,23 @@ void DXRBase::reloadShaders() {
...
-bool DXRBase::onEvent(Event& event) {
-	auto onResize = [&](WindowResizeEvent& event) {
+bool DXRBase::onEvent(const Event& event) {
+	auto onResize = [&](const WindowResizeEvent& event) {
 		// Window changed size, resize output UAV
 		createInitialShaderResources(true);
 		return true;
 	};
 
-	auto onGameOver = [&](GameOverEvent& event) {
```
Comment (OliverGlandberger) [inline Sail/src/API/DX12/dxr/DXRBase.cpp]: This looks suspicious, the fact that 'onGameOver' is completely removed and not replaced with a modified version. If this is intended, ignore this comment.

File: SPLASH/src/game/events/NetworkWelcomeEvent.h (hunk 1 above)
Comment (Smaugmuel) [reply SPLASH/src/game/events/NetworkWelcomeEvent.h]: Moved to public , therefore no "m_".
Replaced with a const list as well.

File: Sail/src/API/DX12/dxr/DXRBase.cpp (hunk 2 above)
Comment (Smaugmuel) [reply Sail/src/API/DX12/dxr/DXRBase.cpp]: DXRBase does not care about when the game is over, it just wants to know when to reset the water.
Game over event replaced by reset water event in this file.

**Pull request merged**